tests/core/test_variable.py

```diff
@@ -0,0 +1,558 @@
+        dx = x.derivative(u)
+        self.assertAlmostEqual(dx, 0.0)
+
+        #test derivative with 2 levels of dependency
+        y = DepVar(w=w, x=x)
+        dy = y.derivative(t)
+        self.assertAlmostEqual(dy, 0.0)
+        dy = y.derivative(u)
+        self.assertAlmostEqual(dy, -0.25)
+        dy = y.derivative(v)
+        self.assertAlmostEqual(dy, 0.0)
+
+        #test derivative with more complex dependency
+        z = DepVar(t=t, w=w, y=y)
+        dz = z.derivative(u)
+        self.assertAlmostEqual(dz, -0.375)
+        dz = z.derivative(v)
+        self.assertAlmostEqual(dz, -0.25)
+        dz = z.derivative(w)
+        self.assertAlmostEqual(dz, 0.75)
+
+        #test derivative with 'multiple' dependency on same object
+        q = DepVar(t=t, u=t, v=t)
+        dq = q.derivative(t)
+        self.assertAlmostEqual(dq, -1.5)
+
+        q = DepVar(t=t, u=t, v=v, w=v)
+        dq = q.derivative(t)
+        self.assertAlmostEqual(dq, -1.0)
+        dq = q.derivative(v)
+        self.assertAlmostEqual(dq, -1.0)
+
+        q = DepVar(w=x, x=x)
+        dq = q.derivative(t)
+        self.assertAlmostEqual(dq, 0.5)
+
+        #test derivative with circular dependencies
+        a = DepVar(t=t)
+        a.t = DepVar(t=a)
+        with self.assertRaises(RuntimeError):
+            a.derivative(t)
+
+        a = DepVar(t=t)
+        b = DepVar(t=a)
+        c = DepVar(t=b)
+        a.t = DepVar(t=c)
+        with self.assertRaises(RuntimeError):
+            a.derivative(t)
+
+class test_SameAs(unittest.TestCase):
+    """Unit tests for relentless.SameAs"""
+
+    def test_init(self):
+        """Test creation with data."""
+        #test variable dependent on DesignVariable
+        v = relentless.DesignVariable(value=1.0)
+        w = relentless.SameAs(v)
+        self.assertEqual(w.value, 1.0)
+        self.assertCountEqual(w.params, ('a',))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'a':v})
+
+        #test variable dependent on DependentVariable
+        u = relentless.SameAs(w)
+        self.assertEqual(u.value, 1.0)
+        self.assertCountEqual(u.params, ('a',))
+        self.assertDictEqual({p:v for p,v in u.depends}, {'a':w})
+
+        #test invalid variable dependence
+        with self.assertRaises(TypeError):
+            u = relentless.SameAs(1.0)
+
+    def test_value(self):
+        """Test setting values."""
+        #create "chained" dependent variables
+        v = relentless.DesignVariable(value=1.0, low=0.5, high=1.5)
+        w = relentless.SameAs(v)
+        x = relentless.SameAs(w)
+
+        self.assertEqual(v.value, 1.0)
+        self.assertEqual(w.value, 1.0)
+        self.assertEqual(x.value, 1.0)
+
+        #change value of v
+        v.value = 0.4
+        self.assertEqual(v.value, 0.5)
+        self.assertEqual(w.value, 0.5)
+        self.assertEqual(x.value, 0.5)
+
+    def test_derivative(self):
+        """Test _derivative method."""
+        v = relentless.DesignVariable(value=1.0)
```

Comment:
Let's use a different value than the derivative, just in case there's a typo.
```suggestion
        v = relentless.DesignVariable(value=2.0)
```
Can you also add one line where you change the value and check the deriv again?